.github/workflows/ci.yml

```
@@ -0,0 +1,106 @@
+name: RIT Action Continuous Integration Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  packages: write
+env:
+  TEST_TAG:  ${{ github.event.repository.name }}/rit:test
+  LATEST_TAG: ghcr.io/rsksmart/${{ github.event.repository.name }}/rit
+
+jobs:
+  build-push-rit-action-image:
+    name: Test  RIT Action docker image
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ env.LATEST_TAG }}
+            ${{ env.TEST_TAG }}
+
+      - name: Setup Docker BuildX
+        id: setup-buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+          driver-opts: network=host
+          platforms: linux/amd64
+
+      - name: Build and export locally Docker
+        uses: docker/build-push-action@v6
+        with:
+          context: .github/images
+          load: true
+          tags: ${{ env.TEST_TAG }}
+
+      - name: Test the RIT Container Action
+        id: test-container
+        env:
+          RSKJ_BRANCH: 'master'
+          FEDERATOR_BRANCH: 'master'
+        run: |
+          docker run \
+            --env RSKJ_BRANCH="${{ env.RSKJ_BRANCH }}" \
+            --env FEDERATOR_BRANCH="${{ env.FEDERATOR_BRANCH }}" \
```

1. what's the point of these vars?
```suggestion
        run: |
          docker run \
            --env RSKJ_BRANCH="master" \
            --env FEDERATOR_BRANCH="master" \
```

2. is it suppose to be only master? shouldn't we test it on the PRs? I think I'm missing something...

---

To configure the branch using the actions input variable. By default, we run using master, but imagine that you want to test from a branch, or from a tag, you just call the action setting it and the test will run on that branch or tag.

---

1. This var it's making this container-action support configuration from the branch of RSKJ and POWPEG that you want to validate with your action. Imagine that you are executing the action built here by this CI on RSKJ, you can use a variable on this input executing the RIT on the branch or tag passed through this variable.

2. It can be anything, can be RC, tags, branches...I am creating the action flexible enough to support any input to be used on the checkout. Be aware that this CI.yml, it's responsible to build and push the `RIT container action` this action will be the one added on the powpeg and rskj pipelines.

---

no, sorry I didn't explain myself very well. What I meant was that why hardcode an env var to the step just to use is as var next. Could you not just use the hardcoded values directly?